### PR TITLE
Change conditions of IP address checking in vm_get_ip.yml

### PR DESCRIPTION
**Pull request by Tomorrow9**

### CODE REVIEW

File: common/vm_get_ip_esxcli.yml [hunk 1]
```diff
@@ -28,6 +28,7 @@
   until:
     - get_vm_ip_esxcli.stdout is defined
     - get_vm_ip_esxcli.stdout
+    - get_vm_ip_esxcli.stdout | from_yaml
```
Comment (123lzxm) [inline common/vm_get_ip_esxcli.yml:31]: The reason for adding the condition?

**Review by 123lzxm — DISMISSED** [review]

File: common/vm_get_ip.yml [hunk 2]
```diff
@@ -48,6 +48,8 @@
 - name: "Check VM '{{ vm_name }}' IP address"
   ansible.builtin.assert:
     that:
-      - vm_guest_ip is defined and vm_guest_ip
-    fail_msg: "Failed to get VM '{{ vm_name }}' IP Address"
+      - vm_guest_ip
+      - vm_guest_ip | ansible.utils.ipaddr
+      - vm_guest_ip != "0.0.0.0"
```
Comment (keirazhang) [inline common/vm_get_ip.yml:53]: 169.254.x.x is not a valid IP either.

**Review by keirazhang — APPROVED** [review]

File: common/vm_get_ip.yml (hunk 2 above)
Comment (Tomorrow9) [reply common/vm_get_ip.yml:53]: Add it, thanks.

File: common/vm_get_ip_esxcli.yml (hunk 1 above)
Comment (Tomorrow9) [reply common/vm_get_ip_esxcli.yml]: That line can be removed. Thanks.

**Pull request merged**